src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -134,26 +134,65 @@ export const LedgerCryptoProvider: (transport: Transport) => Promise<CryptoProvi
     }
   })
 
-  const prepareChangeOutput = (
-    lovelaceAmount: TxTypes.Uint,
-    changeOutput: _AddressParameters,
-    tokenBundle: LedgerTypes.AssetGroup[] | null,
-    datumHashHex?: string,
-  ): LedgerTypes.TxOutput => ({
-    destination: {
-      type: LedgerTypes.TxOutputDestinationType.DEVICE_OWNED,
-      params: {
-        type: changeOutput.addressType,
+  const prepareDestination = (
+    address: Buffer,
+    changeAddressParams: _AddressParameters | null,
+    signingMode: SigningMode,
+  ): LedgerTypes.TxOutputDestination => {
+    if (changeAddressParams && areAddressParamsAllowed(signingMode)) {
+      return {
+        type: LedgerTypes.TxOutputDestinationType.DEVICE_OWNED,
         params: {
-          spendingPath: changeOutput.paymentPath as BIP32Path,
-          stakingPath: changeOutput.stakePath,
+          type: changeAddressParams.addressType,
+          params: {
+            spendingPath: changeAddressParams.paymentPath as BIP32Path,
+            stakingPath: changeAddressParams.stakePath,
```

Comment:
I don't see why one has `as BIP32Path` and the other not.

Reply:
I think it's because `paymentPath` is always defined if `changeAddressParams` exist and `stakePath` can be undefined in the case of an enterprise address. I removed it and added an "assertion" with a comment in https://github.com/vacuumlabs/cardano-hw-cli/pull/127/commits/1b2aaf0d83967b6a20dd64ab946afef98f23361e#diff-f91c370b4a0d3aba3dde7c9ac1af1e0a2a643f066593823207cc9620e150cbc8R144.